From cc40dc2d1920ef9dabcfeea90f5864bec1422402 Mon Sep 17 00:00:00 2001
From: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
Date: Thu, 6 Dec 2018 21:22:32 +0530
Subject: [PATCH] Fix for BT reset issue

Intel BT Controller can not handle 2 Commands from the host
at the same time resulting in unstatable Bluetooth.

Fix the issue by reducing the command credit in all cases equal
to 1.

Test: Pairing/Connection/File Transfer/A2DP found to be working.

Change-Id: I204f2c10b63d54803482f43d07db1b275ee8cfd5
Tracked-On: OAM-72658
Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
---
 bluetooth/1.0/default/Android.bp     | 4 ++++
 bluetooth/1.0/default/h4_protocol.cc | 9 ++++++++-
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/bluetooth/1.0/default/Android.bp b/bluetooth/1.0/default/Android.bp
index 48bbadf8a..f1a285bcc 100644
--- a/bluetooth/1.0/default/Android.bp
+++ b/bluetooth/1.0/default/Android.bp
@@ -56,6 +56,10 @@ cc_library_static {
     name: "android.hardware.bluetooth-hci",
     vendor: true,
     defaults: ["hidl_defaults"],
+    include_dirs: [
+        "system/bt/device/include",
+        "system/bt/stack/include"
+    ],
     srcs: [
         "hci_packetizer.cc",
         "hci_protocol.cc",
diff --git a/bluetooth/1.0/default/h4_protocol.cc b/bluetooth/1.0/default/h4_protocol.cc
index 318ba45a1..32ce276a2 100644
--- a/bluetooth/1.0/default/h4_protocol.cc
+++ b/bluetooth/1.0/default/h4_protocol.cc
@@ -23,7 +23,7 @@
 #include <log/log.h>
 #include <sys/uio.h>
 #include <unistd.h>
-
+#include <hcidefs.h>
 namespace android {
 namespace hardware {
 namespace bluetooth {
@@ -112,6 +112,13 @@ void H4Protocol::OnDataReady(int fd) {
           LOG_ALWAYS_FATAL("%s: Unimplemented packet type %d", __func__,
                            static_cast<int>(hci_packet_type_));
         } else {
+	    if(tpkt.data()[1] == HCI_COMMAND_COMPLETE_EVT) {
+                ALOGV("%s Command complete event ncmds = %d", __func__, tpkt.data()[3]);
+                tpkt.data()[3] = 1;
+            } else if (tpkt.data()[1] ==  HCI_COMMAND_STATUS_EVT) {
+                ALOGV("%s Command status event ncmd = %d", __func__, tpkt.data()[4]);
+                tpkt.data()[4] = 1;
+            }
             hci_packetizer_.CbHciPacket(tpkt.data() + 1, bytes_read - 1);
         }
     }
--
2.17.1

